docker-compose.yml

```diff
@@ -42,9 +32,4 @@ services:
     working_dir: /code/wagtail
     volumes:
       - ./wagtail:/code/wagtail:delegated,rw
-    command:
-      - /bin/sh
-      - -c
-      - |
-        cp -r /node_modules /code/wagtail
-        npm run start
+    command: npm run start
```

Comment:
🤔 This didnt work for me at first. My frontend container either didnt start, or started and died pretty quickly. 

I ended up running `npm i` and `npm run start` on my machine outside of Docker, and restarting the containers to get the static assets to load for me. 

At first I thought this was a problem, but it might not be. Running npm inside of a Docker container on a Mac is super CPU intensive and other than caching the node modules Ive never ben able to figure out how to reduce the CPU % from a frontend container. 
![image](https://user-images.githubusercontent.com/4743971/73869053-d5ab5300-4806-11ea-92d0-be49ccbf62db.png)

It might be worth having a conversation about running npm outside of the containers, or if that level of separation isn't necessary since Wagtail is largely a backend application.

Reply:
Confirmed, the frontend container dies. I didn't look into it more than the error messages it produces though. 

```
frontend_1  | > wagtail@1.0.0 start /code/wagtail   
frontend_1  | > npm run watch         
frontend_1  | > wagtail@1.0.0 watch /code/wagtail   
frontend_1  | > npm-run-all --parallel gulp:dev:watch webpack:dev:watch         
frontend_1  | sh: 1: npm-run-all: not found         
frontend_1  | npm ERR! code ELIFECYCLE
frontend_1  | npm ERR! syscall spawn  
frontend_1  | npm ERR! file sh        
frontend_1  | npm ERR! errno ENOENT   
frontend_1  | npm ERR! wagtail@1.0.0 watch: `npm-run-all --parallel gulp:dev:watch webpack:dev:watch`       
frontend_1  | npm ERR! spawn ENOENT   
frontend_1  | npm ERR!  
frontend_1  | npm ERR! Failed at the wagtail@1.0.0 watch script.  
frontend_1  | npm WARN Local package.json exists, but node_modules missing, did you mean to install?        
frontend_1  | npm ERR! A complete log of this run can be found in:
frontend_1  | npm ERR!     /root/.npm/_logs/2020-02-05T18_09_37_587Z-debug.log  
frontend_1  | npm ERR! code ELIFECYCLE
frontend_1  | npm ERR! errno 1        
frontend_1  | npm ERR! wagtail@1.0.0 start: `npm run watch`       
frontend_1  | npm ERR! Exit status 1  
frontend_1  | npm ERR!  
frontend_1  | npm ERR! Failed at the wagtail@1.0.0 start script.  
frontend_1  | npm WARN Local package.json exists, but node_modules missing, did you mean to install?        
frontend_1  | npm ERR! A complete log of this run can be found in:
frontend_1  | npm ERR!     /root/.npm/_logs/2020-02-05T18_09_37_598Z-debug.log
```

Running `npm run start` on my machine (outside of Docker) though worked perfectly.

Reply:
Yeah the npm part definitely needs a bit more work. There is a certain requirement that you should be able to build the frontend assets using the container as a build environment to reduce the need for dependencies on your own computer. The wagtail admin frontend relies more and more on ReactJS and of course SCSS so if we can find a way to run this then that would be optimal. The error you were experiencing was probably about the node_modules folder missing at the start.  I'm going to try to work the node_modules install into the Dockerfile for the frontend to reduce the initial start time.